### PR TITLE
[DS-147] Update API in rssh and refactored much common code to lib.sh

### PR DESCRIPTION
**Pull request by Digital-Larry**

### CODE REVIEW

File: p2p/rssh.sh [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+        ;;
+    esac
+done
+
+# get rid of the just-finished flag arguments
+shift $(($OPTIND-1))
+
+# make sure we have something to connect to
+if [ $# -eq 0 -a "$LIST_ONLY" -ne 1 ]; then
+echo "boop"
```
Comment (danawoodman) [inline p2p/rssh.sh]: What is "boop"?

File: p2p/rssh.sh [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+}
+
+
+###############################
+# Main program starts here    #
+###############################
+#
+# Create the config directory if not there
+#
+echo "remote.it rssh.sh Version $VERSION $MODIFIED"
```
Comment (danawoodman) [inline p2p/rssh.sh]: Should we echo this on every call to the command line? Seems excessive and atypical

**Review by danawoodman — COMMENTED** [review]
The readme should have documentation on how to use these tools, we should have basic usage examples and install process.

This will break with large service lists (lists everything from the API /devices/all endpoint). We should make a note about that.

Should use auth hash instead of login. Username/password is less than ideal (slow and less secure)

Doesn't work on Mac yet, should note that.

File: p2p/rssh.sh (hunk 1 above)
Comment (Digital-Larry) [reply p2p/rssh.sh]: it's a little sound effect I thought I had deleted.

File: p2p/rssh.sh (hunk 2 above)
Comment (Digital-Larry) [reply p2p/rssh.sh]: I can have that only when using -v if you like.

File: p2p/README.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
+p2p-init.sh: remote.it Peer to Peer (P2P) initiator sample script.
```
Comment (danawoodman) [inline p2p/README.md]: Why is this not in the root directory?

Also, we should use proper markdown formatting like we do in our other readmes. Can you add headings, code foramtting, etc to the readme?

Comment (Digital-Larry) [reply p2p/README.md]: You mean of command line repo?  I had no idea how many items we might eventually get here so I made a folder "p2p" for p2p related things, as it turns out this is just one file.  So maybe that was stupid.  Let me look at marking up/down the readme.

Comment (Digital-Larry) [reply p2p/README.md]: p2p-init.sh is now in the root folder.  the README.md contains info about p2p-init.sh and if we add some other commands later we can decide how to handle it at that time.

**Review by danawoodman — APPROVED** [review]

**Pull request merged**